library/xml-model/src/main/java/org/cru/godtools/xml/model/AnalyticsEvent.kt

```diff
@@ -47,15 +50,20 @@ class AnalyticsEvent {
     val attributes: Map<String, String>
 
     @RestrictTo(RestrictTo.Scope.TESTS)
-    constructor(delay: Int = 0) {
-        action = null
+    constructor(
+        action: String? = null,
+        delay: Int = 0,
+        systems: Set<AnalyticsSystem>? = null,
+        attributes: Map<String, String>? = null
+    ) {
+        this.action = action
         this.delay = delay
-        systems = emptySet()
+        this.systems = systems.orEmpty()
```

Comment:
This just returns an empty string if there is no other string value right?

Reply:
empty set because `systems` is a `Set<AnalyticsSystem>`

Reply:
Aahh gotcha its just returning an empty value for whatever the type is. Got it!